### PR TITLE
style: new design colors, spacing and layout

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Implement an extrapolated version of the new design provided by the designer without changes in the intrinsic behavior of the components.

# Changes

- review all theming CSS
- introduce a new `Content` component
- review `Tag` component
- few new assets (imported as such and resolved by Vite automatically - no conversion to `.svelte` cmp) and icons
- update jest config to load new svg and png assets

# Screenshots

No screenshots. Test version provided internally.

### CODE REVIEW

File: jest-transform.cjs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const path = require("path");
```
Comment (lmuntaner) [inline jest-transform.cjs:1]: Why is this needed now? It wasn't needed before for svg assets right?

File: src/docs/styles/docs-global.scss [hunk 2]
```diff
@@ -56,12 +56,21 @@ main {
...
     margin: var(--padding-3x) 0;
   }
+
+  table {
+    background: var(--card-background);
+  }
+
+  pre[class*="language-"] {
+    // We always want dark code block
+    background: #34294e;
```
Comment (lmuntaner) [inline src/docs/styles/docs-global.scss:66]: Can we use a variable here?

File: src/lib/components/Content.svelte [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+  export let back = false;
+
+  // Enhance UI contrast for readability
+  export let contrast = false;
+</script>
+
+<div
+  class="content"
+  style={`--layout-bottom-offset: ${$layoutBottomOffset}px`}
```
Comment (lmuntaner) [inline src/lib/components/Content.svelte:15]: Why is this variable set inline?

**Review by lmuntaner — COMMENTED** [review]
Just a few comments, leaving the "real" CR for @mstrasinskis  😄

File: jest-transform.cjs (hunk 1 above)
Comment (peterpeterparker) [reply jest-transform.cjs:1]: Because this PR uses assets https://github.com/dfinity/gix-components/tree/style/new-design-colors-spacing-layout/src/lib/assets

Not svg encapsulated as svelte component but real assets which are `import ... from ....svg` with Vite and therefore are also resolved when the lib is packaged.

File: src/docs/styles/docs-global.scss (hunk 2 above)
Comment (peterpeterparker) [reply src/docs/styles/docs-global.scss:66]: I did not find any passing existing variable in light mode and did not wanted to add a variable in the lib as this code block is only meant for the doc showcase - should not exists in the lib. That's why I used a fix value and added a comment.

File: src/lib/components/Content.svelte (hunk 3 above)
Comment (peterpeterparker) [reply src/lib/components/Content.svelte:15]: Because the CSS style needs a dynamic binded value calculated with JavaScript. Not sure I get your question, how would you solve this differently?

Note: this is not new but refactored.

Comment (lmuntaner) [reply src/lib/components/Content.svelte:15]: I don't know, but this is the 8th way we have to change style.

https://www.notion.so/Reusing-styles-06430b86e1f14e698dd7b44a99d2b6c2

Maybe this is the only way to fix it.

I also think we should push back designing some things if they add a big layer of complexity.

Comment (peterpeterparker) [reply src/lib/components/Content.svelte:15]: Haha I still don't get your point here 😅. Not sure it is worth investing that deeper here, if you want to discuss it in a chat ping me anytime.

Comment (lmuntaner) [reply src/lib/components/Content.svelte:15]: No point for this PR, no worries. I am just not a fan of having so many ways of changing styles, and we added one more. Maybe it was the only way here. Just a comment to bring that topic, but I don't think you and @mstrasinskis are worried about that, so that's good.

Comment (peterpeterparker) [reply src/lib/components/Content.svelte:15]: We have developed this particular piece of code together with @mstrasinskis 😜

File: src/lib/components/Modal.svelte [hunk 4]
```diff
@@ -27,7 +27,7 @@
 {#if visible}
   <div
     class="modal"
-    transition:fade
+    transition:fade={{ duration: 150 }}
```
Comment (mstrasinskis) [inline src/lib/components/Modal.svelte:30]: Can any of 2 other values be applied here, to have the same ani.time everywhere, or it looks better with 150?
```
  --animation-time-normal: 0.2s;
  --animation-time-short: 0.125s;
  ```

File: src/lib/components/Back.svelte [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <script lang="ts">
-  import IconBackIosNew from "$lib/icons/IconBackIosNew.svelte";
+  import IconBack from "../icons/IconBack.svelte";
```
Comment (mstrasinskis) [inline src/lib/components/Back.svelte:2]: `$lib/`?

File: src/lib/components/Modal.svelte (hunk 4 above)
Comment (peterpeterparker) [reply src/lib/components/Modal.svelte]: Sure 125 works out

File: src/lib/icons/IconPassword.svelte [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<!-- source: DFINITY foundation -->
+<svg
+  width="20"
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply src/lib/icons/IconPassword.svelte]: As discussed, designer choice
